src/megachatapi_impl.cpp

```diff
@@ -793,8 +804,19 @@ void MegaChatApiImpl::sendPendingRequests()
                 request->setChatHandle(chatid);
                 request->setText(room->titleString().c_str());
                 request->setNumber(room->getNumPeers());
+                request->setUserHandle(ph);
+
+                MegaChatErrorPrivate *megaChatError;
+
+                if (room->isActive())
```

Comment:
The special condition here is `if (!room->isActive())`, in which case we want to notify the app in a special way. It's better to honor that condition, rather than changing it for the `ok` case. I would write it as:
```
if (!room->isActive())  // a public chat already left cannot be previewed: join it directly.
{
    megaChatError = new MegaChatErrorPrivate(MegaChatError::ERROR_ACCESS);
}
else
{
    megaChatError = new MegaChatErrorPrivate(MegaChatError::ERROR_OK);
}
```

Reply:
It's deprecated because of rejoin